tap-agent/src/tap/receipt_checks_adapter.rs

```diff
@@ -0,0 +1,205 @@
+        escrow_subgraph_polling_interval: u64,
+    ) -> Eventual<bool> {
+        #[derive(serde::Deserialize)]
+        struct AllocationResponse {
+            #[allow(dead_code)]
+            id: String,
+        }
+
+        #[derive(serde::Deserialize)]
+        struct TransactionsResponse {
+            transactions: Vec<AllocationResponse>,
+        }
+
+        timer(Duration::from_secs(escrow_subgraph_polling_interval)).map_with_retry(
+            move |_| async move {
+                let response = escrow_subgraph
+                    .query::<TransactionsResponse>(&json!({
+                        "query": r#"
+                                    query (
+                                        $sender_id: ID!, 
+                                        $receiver_id: ID!, 
+                                        $allocation_id: String!
+                                    ) {
+                                        transactions(
+                                            where: {
+                                                and: [
+                                                    { type: "redeem" }
+                                                    { sender_: { id: $sender_id } }
+                                                    { receiver_: { id: $receiver_id } }
+                                                    { allocationID: $allocation_id }
+                                                ]
+                                            }
+                                        ) {
+                                            allocationID
+                                            sender {
+                                                id
+                                            }
+                                        }
+                                    }
+                                "#,
+                        "variables": {
+                            "sender_id": sender_address.to_string(),
+                            "receiver_id": indexer_address.to_string(),
+                            "allocation_id": allocation_id.to_string(),
+                        }
+                    }))
+                    .await
+                    .map_err(|e| e.to_string())?;
+                let response = response.data.ok_or_else(|| {
+                    format!(
+                        "No data found in escrow subgraph response for allocation {} and sender {}",
+                        allocation_id, sender_address
+                    )
+                })?;
+                Ok(!response.transactions.is_empty())
+            },
+            move |error: String| {
+                error!(
+                    "Failed to check the escrow redeem status for allocation {} and sender {}: {}",
+                    allocation_id, sender_address, error
+                );
+                sleep(Duration::from_secs(escrow_subgraph_polling_interval).div_f32(2.))
```

Comment:
this should be in `ms`?

Reply:
Fixed in 8e6460f1b86ba67c1846fde6d5ae3345dbcc047c